flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssigner.java

```diff
@@ -275,9 +275,9 @@ private void splitTable(TableId nextTable) {
 
     @Override
     public Optional<MySqlSplit> getNext() {
-        checkSplitterErrors();
         waitTableDiscoveryReady();
         synchronized (lock) {
+            checkSplitterErrors();
```

Comment:
We should also make the assignment logic of the `uncaughtSplitterException` guarded by the `notificationLock` in the method `splitChunksForRemainingTables`.

Reply:
hi @ruanhang1993 I've already modified it,Please help to review it when you have time,thanks.